addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthenticationDiagnostics.java

```diff
@@ -318,14 +319,59 @@ private class ZestClientScreenshotDiag extends ZestClientScreenshot {
 
         private String description;
 
+        private ZestClientElement element;
+
+        public void setClientElement(ZestClientElement element) {
+            this.element = element;
+        }
+
         public void setDescription(String description) {
             this.description = description;
         }
 
         @Override
         public String invoke(ZestRuntime runtime) throws ZestClientFailException {
-            recordStep(runtime.getWebDriver(this.getWindowHandle()), description);
+            recordStep(
+                    runtime.getWebDriver(this.getWindowHandle()),
+                    description,
+                    getWebElement(runtime, element));
             return null;
         }
+
+        private WebElement getWebElement(ZestRuntime runtime, ZestClientElement element) {
+            if (element == null) {
+                return null;
+            }
+
+            WebDriver wd = runtime.getWebDriver(this.getWindowHandle());
+            if (wd == null) {
+                return null;
+            }
+            String elem = runtime.replaceVariablesInString(element.getElement(), false);
+
+            try {
```

Comment:
I think we have this code block in a set of places? Worth a helper method?
I think I've created one, but not PRed the related changes ;)

Reply:
Once we have the newer version of Zest here it can be replaced with the direct call to the statement.